packages/framework/src/Model/Product/ProductInputPriceDataFactory.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroupFacade;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductInputPriceFacade;
+
+class ProductInputPriceDataFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductInputPriceFacade $productInputPriceFacade
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade $vatFacade
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroupFacade $pricingGroupFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly ProductInputPriceFacade $productInputPriceFacade,
+        protected readonly VatFacade $vatFacade,
+        protected readonly PricingGroupFacade $pricingGroupFacade,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductInputPriceData
+     */
+    protected function createInstance(): ProductInputPriceData
+    {
+        return new ProductInputPriceData();
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat $vat
+     * @param array<int, \Shopsys\FrameworkBundle\Component\Money\Money|null> $manualInputPricesByPricingGroupId
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductInputPriceData
+     */
+    public function create(Vat $vat, array $manualInputPricesByPricingGroupId): ProductInputPriceData
+    {
+        $productInputPriceData = $this->createInstance();
+        $productInputPriceData->vat = $vat;
+        $productInputPriceData->manualInputPricesByPricingGroupId = $manualInputPricesByPricingGroupId;
+
+        return $productInputPriceData;
+    }
+
+    /**
+     * @return array<int, \Shopsys\FrameworkBundle\Model\Product\ProductInputPriceData>
+     */
+    public function createEmptyForAllDomains(): array
+    {
+        $productInputPriceData = [];
+        $allPricingGroups = $this->pricingGroupFacade->getAll();
```

Comment:
you can move this inside `getNullForPricingGroupsByDomainId()` method and hence get rid of the `$allPricingGroups` parameter

Reply:
...and maybe, it would make sense to use `pricingGroupFacade->getByDomainId()` there to get rid of the `$pricingGroup->getDomainId() === $domainId` condition, what do you think?

Reply:
I deliberately made it this way because the query to get all the pricing groups is now called just once instead of for each domain. We can discuss it further :)

Reply:
ok, makes sense :slightly_smiling_face: :+1: